shprofile.sh

```diff
@@ -282,11 +309,111 @@ function shpClearEnvironment {
     unset SHP_INVALID_PROFILE
     unset SHP_INVALID_PROFILES
     unset SHP_INVALID_PROFILE_EXECUTION_TYPE
+    unset SHP_PROFILE_CREATED
+    unset SHP_PROFILE_DELETED
+    unset SHP_PROFILE_DELETE_ABORTED
     unset shpIsInformationMessagesDisplayed
     unset shpRequiredProfile
 }
 
-# Parse user-given options and directly execute "self-contained" options
+# Creates a new shprofile
+#
+# @param $@ profile
+function shpCreateProfile {
+    local profileName="$1"
+    if [ -z "$profileName" ]; then
+        shpLog $SHP_ERROR "Profile name is required."
+        return $SHP_INVALID_PROFILE
+    fi
+    local profileHome="$SHP_PROFILES_HOME/$profileName"
+    if [ -d "$profileHome" ]; then
+        shpLog $SHP_ERROR "Profile '$profileName' already exists."
+        return $SHP_INVALID_PROFILE
+    fi
+    
+    mkdir -p "$profileHome"
+    chmod +x "$profileHome"
+    cat > "$profileHome"/001-rcm.sh << EOF
+#!/usr/bin/env bash
+# 
+# Load all rcm-managed dotfiles for this shprofile, represented as a tag in rcm
+rcup -t $profileName
+EOF
+    cat > "$profileHome"/999-rcm-unload.sh << EOF
+#!/usr/bin/env bash
+#
+# Remove all rcm-managed dotfiles tagged with the current shprofile
+rcdn -t $profileName
+EOF
+    shpRequiredProfile=$profileName
+    return $SHP_PROFILE_CREATED
+}
+
+# Deletes an existing shprofile
+#
+# @param $@ profile
+function shpDeleteProfile {
+    local profileName="$1"
+    if [ -z "$profileName" ]; then
+        shpLog $SHP_ERROR "Profile name is required."
+        return $SHP_INVALID_PROFILE
+    fi
+    local profileHome="$SHP_PROFILES_HOME/$profileName"
+    if ! [ -d "$profileHome" ]; then
+        shpLog $SHP_ERROR "Profile '$profileName' does not exist."
+        return $SHP_INVALID_PROFILE
+    fi
+    
+    # Prompt user for confirmation
+    echo "Are you sure you want to delete profile '$profileName'? [y/N] "
+    read -r confirmation
+    if [[ "$confirmation" != [yY] ]]; then
+        echo "Aborting profile deletion."
+        return $SHP_PROFILE_DELETE_ABORTED
+    fi
+    
+    # Check if profile to delete is the current one
+    local currentProfile
+    currentProfile=$(shpGetCurrentProfile)
+    
+    if [ "$currentProfile" = "$profileName" ]; then
+        echo "DEBUG: Forgetting current profile '$profileName'"
+        shpForgetCurrentProfile
+    fi
+
+    # Delete profile
+    rm -rf "$profileHome"
+    return $SHP_PROFILE_DELETED
+}
+
+# BEGIN: executeCommand
+# Execute different commands depending on the argument value
+#
+# @param $@ arguments
+function executeCommand {
```

Comment:
What's the meaning of this function vs `shpParseOptions`?
In addition, this function is never used
Finally, this function should be removed of the user context at the end of `shprofile` execution by using `shpClearEnvironment`